src/node_group.hpp

```diff
@@ -3,26 +3,30 @@
 
 #include <list>
 #include <set>
+#include <stdint.h>
+#include "onepath.hpp"
 #include "tile.hpp"
 
 class NodeGroup{
 private:
     //static inline bool cmp(Tile* a, Tile* b) { return ((*a) > (*b)); };
-    std::set<Tile*, TilePCompare> tiles;
-    std::set<NodeGroup *> adjacent;
-    float center_x, center_y, center_z;
+    std::set<const Tile*, TilePCompare> tiles;
+    std::map<NodeGroup *, OnePath> adjacent;
+    float center_x, center_y, center_z; // volumetric center, a weighted average
+    OnePath path_type_; // the path restraints to get form any tile in this Group to any other tile
 public:
     NodeGroup();
-    NodeGroup(Tile* tile);
-    std::set<NodeGroup *> merge_groups(NodeGroup other);
+    NodeGroup(Tile* tile, OnePath path_type);
+    std::map<NodeGroup *, OnePath> merge_groups(NodeGroup other);
     //~NodeGroup();
     void update();
     const std::set<const Tile*, TilePCompare> get_tiles() const;
-    std::set<Tile*, TilePCompare> get_tiles();
-    void add_adjacent(NodeGroup* NG);
+    void add_adjacent(NodeGroup* NG, OnePath path_type);
+    std::set<const NodeGroup *> get_adjacent_clear(int path_type) const;
     void remove_adjacent(NodeGroup* NG);
-    const std::set<NodeGroup *> get_adjacent() const;
-    bool adjacent_to(const NodeGroup* other) const;
+    std::map<NodeGroup *, OnePath> get_adjacent();
+    std::map<const NodeGroup *, OnePath> get_adjacent() const;
```

Comment:
```suggestion
    std::map<NodeGroup *, OnePath> get_adjacent_map();
    std::map<const NodeGroup *, OnePath> get_adjacent_map() const;
```
Better naming.
Why does one allow modification, and that's the one that DOESN'T return a copy?